docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+          <li><a href="#Architecting-Store-Dependencies-Example">Architecting Store Dependencies Example</a></li>
+          <li><a href="#Architecture-architect">METHOD: Architecture.architect</a></li>
+          <li><a href="#architectureChain">OBJECT: architectureChain</a>
+            <ol style="list-style-type:lower-alpha">
+              <li><a href="#architectureChain-itNeeds">METHOD: architectureChain.itNeeds</a></li>
+              <li><a href="#architectureChain-itOutputs">METHOD: architectureChain.itOutputs</a></li>
+              <li><a href="#architectureChain-and">METHOD: architectureChain.and</a></li>
+            </ol>
+          </li>
+        </ol>
+      </li>
+    </ol>
+  </li>
+</ol>
+
+##<a id="Premise"></a>Premise [#](#Premise)
+>TuxedoJS provides a novel approach to describing the dependencies between stores. In Flux, store dependencies are defined via the `Dispatcher.waitFor` syntax. This allows stores to define the order at which they will receive events from the dispatcher. While this syntax is workable, we consider it to be flawed due to the fact that it requires each store to separately manage the stores it needs to waitFor. This results in distributed app architectures that are difficult to reason about. For an engineer to understand all the store relationships in an app, he/she would need to step through every store file and draw out the map of stores it is dependent on. TuxedoJS tackles this problem by centralizing waitFor statements and providing a more semantic interface for building waitFor relationships.
+
+##<a id="Implementation"></a>Implementation [#](#Implementation)
+TuxedoJS provides two major innovations for defining store dependencies:
+
+1. It allows you to write all of your store dependencies in a single location so only one file is responsible for managing your app architecture
+2. It allows you to describe your store dependencies in terms of the inputs to and outputs from each of your stores
+
+###<a name="Requiring-Architecture"></a>Requiring Architecture [#](#Requiring-Architecture)
+The Architecture interface is exposed via:
+
+```
+    var Architecture = require('tux/Architecture');
+```
+
+###<a id="Architecting-Store-Dependencies"></a>Architecting Store Dependencies [#](#Architecting-Store-Dependencies)
+Let's take a look at an example. Imagine we want to describe the following network of store dependencies where each store is dependent on the store it points to: [#](#Architecting-Store-Dependencies-Example)
+<a id="Architecting-Store-Dependencies-Example"></a>
+
+```
+    userStore <---privateMessageStore
+        ^               ^
+        |               |
+    roomStore <---------|
+        ^               |
+        |               |
+    threadStore <---unreadStore
+        ^               |
+        |               |
+    messageStore <------|
+```
+
+In Flux we would need to write these dependencies spread out across 6 different files. In TuxedoJS, we can do the following:
```

Comment:
The first sentence has a lot of punch mentioning the six different files and it lacks that in the second section where you're describing why our framework has a leg up. Maybe change the whole thing to something like "In Flux we would need to write these dependencies spread out across six different files, while on the other hand, in TuxedoJS, we can accomplish all of that in just one file. Here's an example demonstrating that:"